client/state/jetpack-connect/selectors.js

```diff
@@ -146,6 +146,19 @@ const getGlobalSelectedPlan = function( state ) {
 	return state.jetpackConnect.jetpackConnectSelectedPlans && state.jetpackConnect.jetpackConnectSelectedPlans[ '*' ];
 };
 
+const getSiteIdFromQueryObject = function( state ) {
+	const authorizationData = getAuthorizationData( state );
+	if ( authorizationData.queryObject && authorizationData.queryObject.client_id ) {
+		return parseInt( authorizationData.queryObject.client_id );
+	}
+	return false;
```

Comment:
I'd expect a `get*` selector to return a `null` and not a boolean `false` if data is not found.